lib/arca.rb

```diff
@@ -0,0 +1,55 @@
+require_relative "arca/collector"
+require_relative "arca/model"
+require_relative "arca/report"
+require_relative "arca/callback_analysis"
+
+module Arca
+
+  # Public: Reader method for accessing the Arca::Model for analysis and
+  # reporting.
+  def self.[](klass)
+    Arca::Model.new(klass)
```

Comment:
May want to guard against string/symbol class or require kind_of?(Class) or something to fail early if someone passes in the wrong type.